amlip_cpp/src/cpp/node/collaborative_learning/ModelManagerReceiverNode.cpp

```diff
@@ -125,50 +126,65 @@ void ModelManagerReceiverNode::process_routine_(
 {
     while (running_)
     {
+start_loop:
+
+        logDebug(AMLIPCPP_NODE_MODELMANAGERRECEIVER, "Waiting for statistics...");
+        statistics_reader_->wait_data_available();
+
+        if (!running_)
+        {
+            goto finish_routine;
+        }
+
         eprosima::amlip::types::ModelStatisticsDataType statistics;
         try
         {
-            logDebug(AMLIPCPP_NODE_MODELMANAGERRECEIVER, "Waiting for statistics...");
-            statistics_reader_->wait_data_available();
-
-            if (!running_)
-            {
-                break;
-            }
-
             // Read statistics
             logDebug(AMLIPCPP_NODE_MODELMANAGERRECEIVER, "Statistics received. Reading...");
             statistics = statistics_reader_->read();
+        }
+        catch (const eprosima::utils::InconsistencyException& e)
+        {
+            std::cerr << e.what() << std::endl;
+            goto start_loop;
```

Comment:
Please use break and continue instructions instead, goto seems to be highly discouraged.